presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSink.java

```diff
@@ -442,11 +466,34 @@ public int[] partitionPage(Page page)
             for (int i = 0; i < columns.size(); i++) {
                 PartitionColumn column = columns.get(i);
                 Block block = page.getBlock(column.getSourceChannel());
+                // adjust legacy timestamp value to compatible with Iceberg non-identity transform calculation
+                if (column.sourceType instanceof TimestampType && session.getSqlFunctionProperties().isLegacyTimestamp() && !column.getField().transform().isIdentity()) {
```

Comment:
I think this conditional here makes the code harder to read. Can we try factoring this out into some other function? Or at least factor out the logic which adjusts the timestamp into a separate function.

Reply:
Sure, thanks for the suggestion, I have extract a separate function `adjustBlockIfNecessary` to strip this part of logic out.